src/Mapbender/CoreBundle/Controller/ApplicationController.php

```diff
@@ -366,26 +375,28 @@ public function instanceTunnelAction($slug, $instanceId)
             $postParams = array_merge($vendorspec, $postParams);
         }
         $proxy_config = $this->container->getParameter("owsproxy.proxy");
-        if (isset($getParams['legendurl'])) {
-            $url = $getParams['legendurl'];
-            unset($getParams['legendurl']);
-        } else {
-            foreach ($getParams as $key => $value) {
-                if (strtolower($key) === 'request') {
-                    // TODO implement for other ogc services
-                    if (strtolower($value) === 'getmap') {
-                        $url = $instance->getSource()->getGetMap()->getHttpGet();
-                    } elseif (strtolower($value) === 'getfeatureinfo') {
-                        $url = $instance->getSource()->getGetFeatureInfo()->getHttpGet();
-                    } elseif (strtolower($value) === 'getlegendgraphic') {
-                        $url = $instance->getSource()->getGetLegendGraphic()->getHttpGet();
-                    }
-                    break;
-                }
+        $requestType = null;
+        foreach ($getParams as $key => $value) {
```

Comment:
Maybe it's better to do it like this:
`$request = $this->getRequest();
$requestType = $request->query->get('request')` ?

Reply:
This is WMS-/OGC-specific quirk where query parameter names are case-insensitive. I.e. some people use REQUEST=GetMap, some people use request=GetMap. If I do request->query->get('request') I will not catch the upper-case variant.

I myself would prefer using a case-insensitive RequestParameterBag implementation to make code like this more readable, and to have this behaviour documented in a central place.